### ISSUE TITLE
[DropDownList] built-in filtering directive does not work with newer Angular version

### ISSUE THREAD

**Issue by mbechev** [telerik/kendo-angular#3201]
**Describe the bug**
The [kendoDropDownFilter ](https://www.telerik.com/kendo-angular-ui/components/dropdowns/dropdownlist/filtering/#toc-built-in-directive)directive does not work as expected in an Angular 10/11 application. 

It seems that the order of registering property values is changed since the filtering works as expected, if the filter directive is defined after **textField** and **valueField** options: 

```html
    <kendo-dropdownlist
      [data]="data"
      [textField]="'text'"
      [valueField]="'value'"
      [kendoDropDownFilter]="filterSettings"
    >
    </kendo-dropdownlist>
```

**Comment by jvalkeejarvi** [comment]
I encountered this bug too and it seems that this can be reproduced only when ivy is enabled. I created a repository for reproducing this issue: https://github.com/jvalkeejarvi/kendo-dropdown-filter-bug

**Comment by dimitar-pechev** [comment]
@jvalkeejarvi, the fix for this is already available in the [development channel](https://www.telerik.com/kendo-angular-ui/components/installation/development-builds/) of the package. It's currently undergoing our QA procedures, however, you can download it and give it a try even now. It's available in `@progress/kendo-angular-dropdowns@5.2.0-dev.202102151644`.

An official release of the package will follow probably in the next couple of weeks,

**Comment by svetq** [comment]
fixed in v`5.2.0` of the @progress/kendo-angular-drodowns package
